### PR TITLE
BRASS-469: Update PingFederate Upgrade Documentation to Work Without …

### PR DESCRIPTION
**Pull request by erikostien-pingidentity**
…PingDownloader

- Update yaml files to use newer image tags
- Remove get-bits.sh, as bits.pingidentity.com is deprecated.
- Update pf-upgrade.sh to expect product zip and license on
the file system instead of retrieving it via get-bits.sh.
- Update documentation to reflect this change in scripts.

### CODE REVIEW

File: 30-helm/pingfederate-upgrade/01-background.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ global:
 
 pingfederate-admin:
   image:
-    tag: 10.2.5-edge
```
Comment (henryrecker-pingidentity) [inline 30-helm/pingfederate-upgrade/01-background.yaml:13]: Any reason to use edge here, vs latest?

File: 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh [hunk 2]
```diff
@@ -1,110 +1,160 @@
...
+    echo "ERROR: No Pingfederate zip file found at file location ${new_pf_zip_path}"
+    exit 1
+fi
+
+# Extract new pingfederate bits
+echo "INFO: Extracting new PingFederate from zip"
+unzip -qd /opt/new "${new_pf_zip_path}"
+test "${?}" -ne 0 && echo "ERROR: Failed to unzip file ${new_pf_zip_path}" && exit 1
+
+# Make sure "pingfederate.lic" has been provided at "/tmp/pingfederate.lic
```
Comment (henryrecker-pingidentity) [inline 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh:72]: Is a new license needed for a minor version upgrade? Like 11.0 to 11.1?

Comment (erikostien-pingidentity) [reply 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh]: I'm not sure this is needed on patch upgrades.

For minor upgrades, the license does specify major.minor version, so I would assume this is needed there. @samir-gandhi Thoughts?

$ pingctl license pingfederate 11.1
ID=00392207
WSTrustSTS=true
OAuth=true
SaasProvisioning=true
Product=PingFederate
Version=11.1
...

File: 30-helm/pingfederate-upgrade/01-background.yaml (hunk 1 above)
Comment (erikostien-pingidentity) [reply 30-helm/pingfederate-upgrade/01-background.yaml:13]: None that I can see, though the hope is that the user is changing this regardless for their upgrade specification.

Comment (erikostien-pingidentity) [reply 30-helm/pingfederate-upgrade/01-background.yaml:13]: Perhaps it might be best to force input here via <image tag here> or something similar

Comment (henryrecker-pingidentity) [reply 30-helm/pingfederate-upgrade/01-background.yaml:13]: Yeah, "image tag here" sounds good. We'll just need to mention that those tags need to be replaced in the doc page.

File: 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh (hunk 2 above)
Comment (samir-gandhi) [reply 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh]: apparently there are scenarios where pf may care about minor. so, better safe than sorry.

File: 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh [hunk 3]
```diff
@@ -1,110 +1,162 @@
...
+    test "${pf_http_response}" -eq 200; then
+    echo "INFO: PingFederate is running, stopping now"
+    pkill /opt/java/bin/java
+    test "${?}" -ne 0 && echo "ERROR: Failed to stop Pingfederate" && exit 1
+
+    # Let PingFederate shutdown
+    sleep 15
 else
-  echo "PF isn't running, continuing"
+    echo "INFO: PingFederate is not running, continuing"
```
Comment (samir-gandhi) [inline 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh:98]: PingFederate no worky, continuing.

File: 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh [hunk 4]
```diff
@@ -1,110 +1,162 @@
...
-  newPfDir=$(dirname "/opt/new/pingfederate-${NEW_PF_VERSION}/pingfederate/${line}")
-  newProfileDir=$(dirname "/opt/new_staging/instance/${line}")
-  mkdir -p "${newPfDir}" "${newProfileDir}"
-  cp "/opt/new/pingfederate-${NEW_PF_VERSION}/pingfederate/${line}" "/opt/new_staging/instance/${line}" > /dev/null 2>&1 
-  diff -q "/opt/staging_bak/instance/${line}" "/opt/new_staging/instance/${line}" >> /tmp/stagingDiffs 2>&1
+    new_pf_dir=$(dirname "/opt/new/pingfederate-${NEW_PF_VERSION}/pingfederate/${line}")
+    new_profile_dir=$(dirname "/opt/new_staging/instance/${line}")
+    mkdir -p "${new_pf_dir}" "${new_profile_dir}"
+    cp "/opt/new/pingfederate-${NEW_PF_VERSION}/pingfederate/${line}" "/opt/new_staging/instance/${line}" > /dev/null 2>&1
+    diff --brief "/opt/staging_bak/instance/${line}" "/opt/new_staging/instance/${line}" >> /tmp/stagingDiffs 2>&1
```
Comment (samir-gandhi) [inline 30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh:143]: nice flag

**Review by henryrecker-pingidentity — APPROVED** [review]

**Comment by cdhesse** [comment]
@henryrecker-pingidentity , @erikostien-pingidentity  - looking at this change, it appears that only the get-bits.sh script required the ping devops-secret... however the pf-upgrade.sh script still requires it....

**Comment by henryrecker-pingidentity** [comment]
@cdhesse We'll get that fixed, thanks for pointing it out.

**Pull request merged**